preload/preconnect-onerror-event.html

```diff
@@ -0,0 +1,37 @@
+<!DOCTYPE html>
+<html>
+<title>Makes sure that preloaded resources trigger the onerror event</title>
+<meta name="timeout" content="long">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/common/get-host-info.sub.js"></script>
+
+<body>
+<script>
+    const {HTTP_REMOTE_ORIGIN} = get_host_info();
+
+    function test_preconnect(origin, resource, desc) {
+        promise_test(async t => {
+            const result = await new Promise(async didLoad => {
+                const href = `${origin}${resource}`;
+                for (const rel of ['preconnect', 'preload']) {
+                    const link = document.createElement('link');
+                    link.href = href;
+                    link.as = 'script';
+                    link.rel = rel;
+                    link.addEventListener('load', () => didLoad({rel, type: 'load'}));
+                    link.addEventListener('error', () => didLoad({rel, type: 'error'}));
+                    document.head.appendChild(link);
+                    await new Promise(resolve => t.step_timeout(resolve, 200));
```

Comment:
So the `preconnect` promises hang? Maybe after they've failed the race against the timeout, reject them and test for that?

Reply:
What do you mean by "hang"? They're simply an event listener that's not fired.

Reply:
Also, the wait needs to happen before calling the preload, to make sure the preload reuses the preconnect if it's there and it's not a race issue

Reply:
I thought we have a promise that doesn't resolve, but that's not the case.
At the same time, it seems like an implementation that fires load/error events for preconnect, but does that later than the same events for preload would also pass here..